src/read.rs

```diff
@@ -62,7 +62,7 @@ impl<R: Read> RngCore for ReadRng<R> {
         if dest.len() == 0 { return Ok(()); }
         // Use `std::io::read_exact`, which retries on `ErrorKind::Interrupted`.
         self.reader.read_exact(dest).map_err(|err| {
-            Error::with_cause(ErrorKind::Unavailable, "ReadRng: read error", err)
```

Comment:
Does it make sense to split this in two errors? To differentiate between unrecoverable errors and others? If I look at [`std::io::ErrorKind`](https://doc.rust-lang.org/std/io/enum.ErrorKind.html) I would only map `:io::ErrorKind::Other` to `ErrorKind::Unexpected`. And all the others seem like `ErrorKind::Unavailable` to me.

Reply:
Good point, but I would think only `NotFound` or `PermissionDenied` would indicate `Unavailable`. If `Read` gets used over the network, any of the refused/reset/aborted *might* indicate that human input is needed to resolve, but it's harder to say; all the rest should be `Unexpected` as I see it.

Reply:
Ah, your right. If we can't imagine the error to occur, best to map it to `Unexpected`. I think besides the 5/6 you mentioned also `BrokenPipe` could also mean `Unavailable`.

Reply:
I don't see how a broken pipe is possible when reading directly from a file so `Unexpected` is fine.

I updated the PR BTW.

Reply:
You can read from `stdin`, just like PractRand does.

Reply:
True. Okay, `BrokenPipe` maps → `Unavailable`